app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+def cache(func: object) -> object:
+    cache = {}
```

Comment:
the cache instance is unique for each function because it is created during decorating the corresponding function

So you can get rid of function name in `signature ` and just use `args`